### DISCUSSION TITLE
Are there recommended tools/MKDocs plugins to make documentation from sources?

### DISCUSSION THREAD

**Discussion by redtide** [mkdocs/mkdocs#3247] [Q&A] [answered]
In particular I'm interested for C++ other than Python.
I'm currently using a Python script that converts Doxygen XML output to JSON, which I use in MKDocs to display on pages via Jinja templates using dedicated CSS.
I'm wondering if there is a tool that already provide this but with more features.

**Comment by oprypin** [comment, marked as answer]
Hi.
The most productive approach would be to develop a Doxygen handler for https://github.com/mkdocstrings/mkdocstrings - I'm sure it would be very welcomed by the community and you can count on support from us in that endeavor. The logic that you're describing that you already have would already get you quite far in terms of what it takes to develop a handler

**Reply by oprypin** [reply]
As a further note, https://github.com/mkdocstrings/crystal is an existing example of a handler where the main logic is performed by parsing JSON produced by an external tool

**Reply by Andre601** [reply]
I was considering to use mkdocsstrings to generate docs for my api written in java.
But that would require me to figure out how Java support could be done, which for me is difficult.

I also don't use gitter, so just joining that chat and asking is out of the question for me personally.

**Reply by redtide** [reply]
Hi! You guys are super fast on replying, and thank you for the input, that sounds good, I'll see what I can do.

**Reply by redtide** [reply]
I'm still busy with other stuff, so I leave here for reference the links to the (incomplete but usable) work I have done if anyone else is interested to collaborate:
- <https://github.com/jekyllize/doxygen/blob/master/scripts/doxygen/doxy2json.py> initial work (license can be changed to bsd).
- <https://github.com/sfztools/sfizz/blob/develop/scripts/doxygen/doxy2json.py> adapted with the help of the main sfizz developer

I forgot the details but I think they should be merged in some way.
The result of the former can be seen at <https://sfz.tools/sfizz/api/>.
~~though there is a visual problem with the dark theme.~~
The site code is at <https://github.com/sfztools/sfztools.github.io>.
~~it's based on Jekyll but there is a mkdocs version in the `mkdocs` folder (with a TODO with the list of missing features for the switch)~~
If anyone is interested feel free to contact/ask me.
